From a1501772832bf453bfdd241d6ff76ef4e1800c62 Mon Sep 17 00:00:00 2001
Message-Id: <a1501772832bf453bfdd241d6ff76ef4e1800c62.1693487783.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Thu, 3 Aug 2023 16:44:16 +0100
Subject: [PATCH] gnu: clisp: Add threads support on x84_64-linux.

Guix's build version of CLISP, fails to load bordeaux-threads:

Error while trying to load definition for system bordeaux-threads
      This implementation is unsupported.

This patch adds support for POSIX_THREADS on x86_64-linux
system.

* gnu/packages/lisp.scm (clisp)[arguments]: Add configuration options to
support threads on x86_64-linux system.

Reported and Co-Authored by Andrew Kravchuk <awkravchuk@gmail.com>
---
 gnu/packages/lisp.scm | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/gnu/packages/lisp.scm b/gnu/packages/lisp.scm
index 2f4fe6476d..867a5df496 100644
--- a/gnu/packages/lisp.scm
+++ b/gnu/packages/lisp.scm
@@ -17,7 +17,7 @@
 ;;; Copyright © 2019-2023 Guillaume Le Vaillant <glv@posteo.net>
 ;;; Copyright © 2020 Marius Bakke <mbakke@fastmail.com>
 ;;; Copyright © 2020 Zhu Zihao <all_but_last@163.com>
-;;; Copyright © 2021 Sharlatan Hellseher <sharlatanus@gmail.com>
+;;; Copyright © 2021, 2023 Sharlatan Hellseher <sharlatanus@gmail.com>
 ;;; Copyright © 2021 Paul A. Patience <paul@apatience.com>
 ;;; Copyright © 2021 Charles Jackson <charles.b.jackson@protonmail.com>
 ;;; Copyright © 2022 Joeke de Graaf <joeke@posteo.net>
@@ -25,6 +25,7 @@
 ;;; Copyright © 2022 ( <paren@disroot.org>
 ;;; Copyright © 2023 Zheng Junjie <873216071@qq.com>
 ;;; Copyright © 2023 Yovan Naumovski <yovan@gorski.stream>
+;;; Copyright © 2023 Andrew Kravchuk <awkravchuk@gmail.com.
 ;;;
 ;;; This file is part of GNU Guix.
 ;;;
@@ -380,6 +381,10 @@ (define-public clisp
                                                      (%current-target-system)))
                                  '("CFLAGS=-falign-functions=4")
                                  '())
+                           ,@(if (target-x86?)
+                                 '("--enable-portability"
+                                   "--with-threads=POSIX_THREADS")
+                                 '())
                             "--with-dynamic-ffi"
                             "--with-dynamic-modules"
                             "--with-ffcall"

base-commit: 782cf09b84bc72514e89d783460ea98bc353bf6a
-- 
2.40.1

